include/sentry.h

```diff
@@ -1186,6 +1186,20 @@ SENTRY_API void sentry_options_add_attachment(
 SENTRY_API void sentry_options_add_attachment_n(
     sentry_options_t *opts, const char *path, size_t path_len);
 
+// TODO add some docstrings
+typedef enum {
+    ATTACHMENT,
+    MINIDUMP,
+    VIEW_HIERARCHY,
+} sentry_attachment_type_t;
+
+SENTRY_API void sentry_options_add_typed_attachment(sentry_options_t *opts,
+    const char *path, sentry_attachment_type_t attachment_type,
+    const char *content_type);
+SENTRY_API void sentry_options_add_typed_attachment_n(sentry_options_t *opts,
+    const char *path, size_t path_len, sentry_attachment_type_t attachment_type,
+    const char *content_type);
```

Comment:
We have [wchar versions ](https://github.com/getsentry/sentry-native/blob/ce9d742d5374d1902f52aa303d776a5d4dee2785/include/sentry.h#L1270-L1276) of `add_attachment`, do we also need this for these new functions?

Reply:
+ `content_type` needs a `content_type_len` parameter for `_n`